core/src/main/java/software/amazon/awssdk/interceptor/ExecutionInterceptor.java

```diff
@@ -0,0 +1,297 @@
+ * <ol>
+ *     <li><i>Override Configuration Interceptors</i> are the most common method for SDK users to register an interceptor. These
+ *     interceptors are explicitly added to the client builder's override configuration when a client is created using the
+ *     {@link software.amazon.awssdk.config.ClientOverrideConfiguration.Builder#addLastExecutionInterceptor(ExecutionInterceptor)}
+ *     method.</li>
+ *
+ *     <li><i>Global Interceptors</i> are interceptors loaded from the classpath for all clients. When any service client is
+ *     created by a client builder, all jars on the classpath (from the perspective of the current thread's classloader) are
+ *     checked for a file named '/software/amazon/awssdk/global/handlers/execution.interceptors'. Any interceptors listed in these
+ *     files (new line separated) are instantiated using their default constructor and loaded into the client.</li>
+ *
+ *     <li><i>Service Interceptors</i> are interceptors loaded from the classpath for a particular service's clients. When a
+ *     service client is created by a client builder, all jars on the classpath (from the perspective of the current thread's
+ *     classloader) are checked for a file named '/software/amazon/awssdk/services/{service}/execution.interceptors', where
+ *     {service} is the package name of the service client. Any interceptors listed in these files (new line separated) are
+ *     instantiated using their default constructor and loaded into the client.</li>
+ * </ol>
+ * </p>
+ *
+ * <p>
+ * <b>Interceptor Order</b>
+ * The order in which interceptors are executed is sometimes relevant to the accuracy of the interceptor itself. For example, an
+ * interceptor that adds a field to a message should be executed before an interceptor that reads and modifies that field.
+ * Interceptor's order is determined by their method of registration. The following order is used:
+ * <ol>
+ *     <li><i>Global Interceptors</i>. Interceptors earlier in the classpath will be placed earlier in the interceptor order than
+ *     interceptors later in the classpath. Interceptors earlier within a specific file on the classpath will be placed earlier in
+ *     the order than interceptors later in the file.</li>
+ *
+ *     <li><i>Service Interceptors</i>. Interceptors earlier in the classpath will be placed earlier in the interceptor order than
+ *     interceptors later in the classpath. Interceptors earlier within a specific file on the classpath will be placed earlier in
+ *     the order than interceptors later in the file.</li>
+ *
+ *     <li><i>Override Configuration Interceptors</i>. Any interceptors registered using
+ *     {@link software.amazon.awssdk.config.ClientOverrideConfiguration.Builder#addLastExecutionInterceptor(ExecutionInterceptor)}
+ *     in the order they were added.</li>
+ * </ol>
+ * When a request is being processed (up to and including {@link #beforeTransmission}, interceptors are applied in forward-order,
```

Comment:
I thought of that exact diagram (it leaves an impression!), but I was worried it would be overkill with this already-long javadoc. Think it's worth including? Most people don't need to worry about the order here as much as they do in netty.